src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -392,55 +477,183 @@ start_profiling(int option, const std::string& port1_name, const std::string& po
   return handle;
 }
 
-uint64_t
+int
 Aie::
-read_profiling(int phdl)
+start_profiling_start_bytes(const std::string& port_name, uint32_t value)
 {
-  uint64_t value = 0;
+  int handle = -1;
 
-  std::vector<Resources::AcquiredResource>& acquiredResourcesForThisHandle = eventRecords[phdl].acquiredResources;
+  XAie_LocType shim_tile;
+  XAie_StrmPortIntf mode;
+  uint8_t stream_id;
 
-  Resources::AcquiredResource& acquiredResource = acquiredResourcesForThisHandle[0];
-  XAie_ModuleType XAieModuleType = AIEResourceModuletoXAieModuleTypeMap[acquiredResource.module];
+  get_profiling_config(port_name, shim_tile, mode, stream_id);
 
-  if (acquiredResource.resource == Resources::performance_counter)
-    XAie_PerfCounterGet(devInst, acquiredResource.loc, XAieModuleType, acquiredResource.id, (u32*)(&value));
-  else
-    throw xrt_core::error(-EAGAIN, "Can't read profiling: The acquired resources order does not match the profiling option.");
+  int handleId = eventRecords.size();
```

Comment:
Here definitely auto should be used.  int and return type doesn't even match in this case.